sdks/java/core/src/main/java/org/apache/beam/sdk/io/util/SerializableAvroCodecFactory.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.sdk.io.util;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
+import static org.apache.avro.file.DataFileConstants.BZIP2_CODEC;
+import static org.apache.avro.file.DataFileConstants.DEFLATE_CODEC;
+import static org.apache.avro.file.DataFileConstants.NULL_CODEC;
+import static org.apache.avro.file.DataFileConstants.SNAPPY_CODEC;
+import static org.apache.avro.file.DataFileConstants.XZ_CODEC;
+
+import java.io.Externalizable;
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.avro.file.CodecFactory;
+
+/**
+ * A wrapper allows  {@link org.apache.avro.file.CodecFactory}s to be serialized using Java's
+ * standard serialization mechanisms.
+ */
+public class SerializableAvroCodecFactory implements Externalizable {
+  private static final long serialVersionUID = 7445324844109564303L;
+  private static final List<String> noOptAvroCodecs = Arrays.asList(NULL_CODEC,
+                                                                    SNAPPY_CODEC,
+                                                                    BZIP2_CODEC);
+  private static final Pattern deflatePattern = Pattern.compile(DEFLATE_CODEC + "-(?<level>-?\\d)");
+  private static final Pattern xzPattern = Pattern.compile(XZ_CODEC + "-(?<level>\\d)");
+
+  private CodecFactory codecFactory;
+
+  // For java.io.Serializable only
+  public SerializableAvroCodecFactory() {}
+
+  public SerializableAvroCodecFactory(CodecFactory codecFactory) {
+    checkNotNull(codecFactory, "Codec can't be null");
+    checkState(checkIsSupportedCodec(codecFactory), "%s is not supported", codecFactory.toString());
```

Comment:
no `.toString()` on stuff using `checkState` -- it will do that automatically in the case that it prints the message, but will also save cycles by not toStringing in the success case.